provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinScriptClassPathProvider.kt

```diff
@@ -152,6 +162,12 @@ class KotlinScriptClassPathProvider(
     val gradleJars by lazy {
         classPathRegistry.getClassPath(gradleApiNotation.name).asFiles
     }
+
+    private
+    val cachedScopeCompilationClassPath = ConcurrentHashMap<ClassLoaderScope, ClassPath>()
+
+    private
+    val cachedClassLoaderClassPath = ClassLoaderClassPathCache()
```

Comment:
Am I reading this correctly? There is a double layer of caching going on here?

Reply:
Yes.